src/main/java/com/configcat/ConfigurationProvider.java

```diff
@@ -55,13 +56,60 @@ public interface ConfigurationProvider extends Closeable {
+     *
+     * @param classOfT     the class of T. Only {@link String}, {@link Integer}, {@link Double} or {@link Boolean} types are supported.
+     * @param key          the identifier of the configuration value.
+     * @param defaultValue in case of any failure, this value will be returned.
+     * @param <T>          the type of the desired config value.
+     * @return the evaluation details.
+     */
+    <T> EvaluationDetails<T> getValueDetails(Class<T> classOfT, String key, T defaultValue);
+
+    /**
+     * Gets the value of a feature flag or setting as T identified by the given {@code key}.
+     *
+     * @param classOfT     the class of T. Only {@link String}, {@link Integer}, {@link Double} or {@link Boolean} types are supported.
+     * @param key          the identifier of the configuration value.
+     * @param user         the user object to identify the caller.
+     * @param defaultValue in case of any failure, this value will be returned.
+     * @param <T>          the type of the desired config value.
+     * @return the evaluation details.
+     */
+    <T> EvaluationDetails<T> getValueDetails(Class<T> classOfT, String key, User user, T defaultValue);
+
+    /**
+     * Gets the value of a feature flag or setting as T asynchronously identified by the given {@code key}.
+     *
+     * @param classOfT     the class of T. Only {@link String}, {@link Integer}, {@link Double} or {@link Boolean} types are supported.
+     * @param key          the identifier of the configuration value.
+     * @param defaultValue in case of any failure, this value will be returned.
+     * @param <T>          the type of the desired config value.
+     * @return a future which computes the the evaluation details.
+     */
+    <T> CompletableFuture<EvaluationDetails<T>> getValueDetailsAsync(Class<T> classOfT, String key, T defaultValue);
+
+    /**
+     * Gets the value of a feature flag or setting as T asynchronously identified by the given {@code key}.
+     *
+     * @param classOfT     the class of T. Only {@link String}, {@link Integer}, {@link Double} or {@link Boolean} types are supported.
+     * @param key          the identifier of the configuration value.
+     * @param user         the user object to identify the caller.
+     * @param defaultValue in case of any failure, this value will be returned.
+     * @param <T>          the type of the desired config value.
+     * @return a future which computes the the evaluation details.
+     */
+    <T> CompletableFuture<EvaluationDetails<T>> getValueDetailsAsync(Class<T> classOfT, String key, User user, T defaultValue);
+
     /**
      * Gets the Variation ID (analytics) of a feature flag or setting synchronously based on its key.
      *
      * @param key                the identifier of the configuration value.
      * @param defaultVariationId in case of any failure, this value will be returned.
      * @return the Variation ID.
      */
+    @Deprecated
```

Comment:
Please provide a deprecation description for each `@Deprecated` attribute about what should be used instead. [For example, this](https://github.com/configcat/kotlin-sdk/pull/5/files#diff-31fa78f052cba1d956fd0a9a671f6d9b927b0e0e262410829d0cae528ad9e95fR121).